frame/nis/src/lib.rs

```diff
@@ -217,12 +217,10 @@ pub mod pallet {
 			+ FunMutate<Self::AccountId>
 			+ FunBalanced<Self::AccountId>
 			+ FunHoldInspect<Self::AccountId>
-			+ FunHoldMutate<Self::AccountId>;
+			+ FunHoldMutate<Self::AccountId, Reason = Self::RuntimeHoldReason>;
 
-		/// The identifier of the hold reason.
-
-		#[pallet::constant]
-		type HoldReason: Get<<Self::Currency as FunHoldInspect<Self::AccountId>>::Reason>;
+		/// Overarching hold reason.
+		type RuntimeHoldReason: Member + MaxEncodedLen + From<HoldReason>;
```

Comment:
```suggestion
		type RuntimeHoldReason: From<HoldReason>;
```
I think `Reason = Self::RuntimeHoldReason` already restricts it enough.

Reply:
Bare minimum would be `TypeInfo` and `Encode`: https://github.com/paritytech/substrate/blob/8548a97f9272c7ca43ac7c091d81e6b9cc8b69e2/frame/support/src/traits/tokens/fungibles/hold.rs#L45

Reply:
Hm weird. For me it compiled with only the `From<HoldReason>` on Rust `1.68.2`/`1.70.0`…  
But its fine either way.